frontend/archive/src/main/java/org/pytorch/serve/archive/model/s3/HttpUtils.java

```diff
@@ -19,20 +23,42 @@ public final class HttpUtils {
     private HttpUtils() {}
 
     /** Copy model from S3 url to local model store */
-    public static void copyURLToFile(URL endpointUrl, File modelLocation, boolean s3SseKmsEnabled)
-            throws IOException {
-        // for a simple GET, we have no body so supply the precomputed 'empty' hash
-        Map<String, String> headers;
-        if (s3SseKmsEnabled) {
-            String awsAccessKey = System.getenv("AWS_ACCESS_KEY_ID");
-            String awsSecretKey = System.getenv("AWS_SECRET_ACCESS_KEY");
-            String regionName = System.getenv("AWS_DEFAULT_REGION");
-            if (!regionName.isEmpty() && !awsAccessKey.isEmpty() && !awsSecretKey.isEmpty()) {
+    public static boolean copyURLToFile(
+            List<String> allowedUrls,
+            String url,
+            File modelLocation,
+            boolean s3SseKmsEnabled,
+            String archiveName)
+            throws FileAlreadyExistsException, IOException, InvalidArchiveURLException {
+        if (ArchiveUtils.validateURL(allowedUrls, url)) {
+            if (modelLocation.exists()) {
+                throw new FileAlreadyExistsException(archiveName);
+            }
+
+            if (archiveName.contains("/") || archiveName.contains("\\")) {
+                throw new IOException(
+                        "Security alert slash or backslash appear in archiveName:" + archiveName);
+            }
+
```

Comment:
there's some more validaiton we can do
* Ensure the url is using the https protocol and not http or ftp
* Potentially check that the IP is not a litteral IP and instead a domain?
* Limit redirections
* ..%2F is used for path traveral attacks
* Optionally some additional checks on user-agent and content-type
* Ensure the url is not a redirect